src/codes.ts

```diff
@@ -43,6 +45,15 @@ const codeList: ReadonlyArray<Readonly<ICode>> = Object.freeze((
 
 export const ChainType = tcomb.irreducible('ChainType', (n) => isValid(n));
 
+const chainCodeFromUnspentType = (u: UnspentType, p = Purpose.external): ChainCode | undefined => {
+  const uTcomb = UnspentTypeTcomb(u);
+  const pTcomb = PurposeTcomb(p);
+  const code = codeList.find(({ type, purpose }) => type === uTcomb && purpose === pTcomb);
+  if (code) {
```

Comment:
I think in most cases we want to throw an error here

Reply:
The reason I didn't want to throw an error here is because there are unspent types that don't correspond to a chain code. Passing a valid unspent type and getting back an error seems like the wrong behavior in my opinion.

Reply:
I changed this to return a `CodesByPurpose` object as suggested below and when that constructor is called with an invalid chain code it throws so this now throws.